src/helpers/credential.rs

```diff
@@ -269,6 +276,149 @@ impl<'a> Credential<'a> {
+            .get_issuer_public_key(&subject_did, "#bbs-key-1")
+            .await?;
+        let use_valid_until= if exp_date.is_some() { true } else { false };
+        let exp_date_str = exp_date.unwrap_or("");
+        let credential_values_str = serde_json::to_string(&credential_subject.data)?;
+
+
+        // create credential offer
+        let context = vec![
+            "https://www.w3.org/2018/credentials/v1".to_string(),
+            "https://schema.org/".to_string(),
+            "https://w3id.org/vc-revocation-list-2020/v1".to_string(),
+        ];
+        let id = Uuid::new_v4().to_string();
+        let r#type = vec!["VerifiableCredential".to_string()];
+        let issuer = subject_did.clone();
+        let valid_until = if use_valid_until {
+            Some(exp_date_str.to_string())
+        } else {
+            None
+        };
+        let date_now = SystemTime::now();
+        let datetime: DateTime<Utc> = date_now.into();
+        let issuance_date = datetime.format("%Y-%m-%dT%TZ").to_string();
+        let credential_subject = CredentialSubject {
+            id: Option::from(subject_did.clone()).map(|s| s.to_owned()), // subject.id stays optional, defined by create_offer call
+            data: schema // fill ALL subject data fields with empty string (mandatory and optional ones)
+                .properties
+                .into_iter()
+                .map(|(name, _schema_property)| (name, String::new()))
+                .collect(),
+        };
+        let credential_schema = CredentialSchemaReference {
+            id: schema.id,
+            r#type: schema.r#type,
+        };
+        let revocation_list_index = credential_revocation_id.unwrap_or(r#"0"#).to_string();
+        let revocation_list_credential = credential_revocation_did.unwrap_or(r#"did:evan:zkp:placeholder_status"#).to_string();
+        let status_id = [revocation_list_credential.clone(), revocation_list_index.clone()].join("#");
+        let credential_status = CredentialStatus {
+            id: status_id,
+            r#type: "RevocationList2020Status".to_string(),
+            revocation_list_index,
+            revocation_list_credential,
+        };
+        let unsigned_credential = UnsignedBbsCredential {
+            context,
+            id,
+            r#type,
+            issuer,
+            valid_until,
+            issuance_date,
+            credential_subject,
+            credential_schema,
+            credential_status,
+        };
+        let unsigned_credential_str = serde_json::to_string(&unsigned_credential)?;
+        let nquads_vc = convert_to_nquads(&unsigned_credential_str).await?;
+
+        let payload = OfferCredentialPayload {
+            issuer: subject_did_str.to_string(),
+            subject: Option::from(subject_did.clone()),
+            nquad_count: nquads_vc.len(),
+        };
+        let offer_str = self
+            .vade_evan
+            .vc_zkp_create_credential_offer(
```

Comment:
Could the offer also be created with the `create_credential_offer` helper function?

Reply:
Not without refactoring...  I'll try in another branch